dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
I'm not too happy with this implementation as `std::atomic<T>` supports any type that is trivially copyable. To check if atomic operations are used instead of locks, [`is_lock_free()`](http://en.cppreference.com/w/cpp/atomic/atomic/is_lock_free) or `is_always_lock_free` (c++17) can be used. We should change the implementation in this way. However this could be done past 0.3.0.